client/internal/dns/server_test.go

```diff
@@ -276,6 +279,96 @@ func TestUpdateDNSServer(t *testing.T) {
 	}
 }
 
+func TestDNSFakeResolverHandleUpdates(t *testing.T) {
+	ov := os.Getenv("NB_WG_KERNEL_DISABLED")
```

Comment:
I am not sure this solution is good pattern. What will be happen if the system run test in parallel? 
On an another branch I solved it with a moc interface. https://github.com/netbirdio/netbird/pull/967/commits/d222c3b3ba84483f0b95e310ed6e21bf6df42f47